util/process_file.py

```diff
@@ -34,3 +35,33 @@ def process_file(session: Session, file: File, callback: Callable[[], None]) ->
         file.processing_error = str(error)
         session.commit()
         raise error
+
+
+def serialize_file(file: File, contents=False) -> Dict[str, str]:
+    result = {
+        "name": file.name,
+        "uploadedAt": isoformat(file.uploaded_at),
+    }
+
+    if contents:
+        result["contents"] = file.contents
+
+    return result
+
+
+def serialize_file_processing(file: File) -> Dict[str, Optional[str]]:
```

Comment:
Think it's worth making a `JSON` type alias?

Reply:
Maybe. I'll tinker with it.

Reply:
The simplest version of type declarations for JSON that are better than `Any` look something like this:

```py
JSONDict = Dict[str, "JSON"]
JSONArray = List["JSON"]
JSONValue = Union[str, int, float, None]
JSON = Union[JSONDict, JSONArray, JSONValue]
```

Unfortunately, [`mypy` does not support recursive types](https://github.com/python/mypy/issues/731). That leaves us with this:

```py
JSONDict = Dict[str, Any]
JSONArray = List[Any]
JSONValue = Union[str, int, float, None]
JSON = Union[JSONDict, JSONArray, JSONValue]
```

I don't think that's really better, so I'm going to just make these `Dict[str, Any]`.

Reply:
What if we just use `JSONDict` as you defined it, since that's mostly what we'll be using it for? I think it will help the readability of the code to know what functions are spitting out data that is intended to be a JSON object.